turnpike/plugins/jwt_oidc.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from flask import request
+from jwt import decode, PyJWTError, PyJWKClient
+from jwt.exceptions import InvalidTokenError
+
+from ..plugin import TurnpikeAuthPlugin
+
+
+class JWTODICAuthPlugin(TurnpikeAuthPlugin):
+    """
+    JWTOIDCAuthPlugin performs authorization on headers that represent a JWT issued by OIDC
+    """
+
+    name = "JWT_OIDC"
+    principal_type = "JWT"
```

Comment:
Wondering if this should be `Associate`

```suggestion
    principal_type = "Associate"
```

Reply:
I think leaving  it as 'JWT" is appropriate here, to support both service accounts and human (Associate) types.
Downstream applications should be checking the JWT payload for further information.